### ISSUE TITLE
Declare StaticArrays functions in StaticArraysCore

### ISSUE THREAD

**Issue by cscherrer** [JuliaArrays/StaticArraysCore.jl#11]
Avoiding a dependency on all of StaticArrays requires being able to define all needed methods using only StaticArraysCore. For example, in [this Accessors PR](https://github.com/JuliaObjects/Accessors.jl/pull/60) we need to be able to this code to work:

```julia
@inline delete(obj::StaticArraysCore.SVector, l::IndexLens) = deleteat(obj, only(l.indices))
@inline insert(obj::StaticArraysCore.SVector, l::IndexLens, val) = insert(obj, only(l.indices), val)
```

But `deleteat` and `insert` are declared (and defined) in StaticArrays itself, so these two lines of code currently require all of StaticArrays before the compiler will even accept them.

This seems to defeat the purpose of having a separation like this in the first place. This makes me wonder, could there be a minimal set of functions in StaticArrays that you'd consider useful for interfacing from "StaticArrays-optional" libraries like Accessors? 

Could `deleteat` and `insert`, and whatever other functions you think might play a similar role, be declared in StaticArraysCore, and then implemented in StaticArray?

**Comment by mateuszbaran** [comment]
Sure, we can declare functions in StaticArraysCore.jl. We already have `similar_type` here, feel free to make a PR for other functions you need.

**Comment by cscherrer** [comment]
Great! Looks like we need `deleteat` and `insert`. I guess we'll need to PR both StaticArraysCore and StaticArrays, to reflect that StaticArraysCore will now "own" both of these functions.

**Comment by mateuszbaran** [comment]
Yes, exactly, one PR like this one: https://github.com/JuliaArrays/StaticArraysCore.jl/pull/5 and one like this: https://github.com/JuliaArrays/StaticArrays.jl/pull/1084
